src/server/routes/query/query.ts

```diff
@@ -56,5 +58,35 @@ export function queryRouter(settings: Pick<SettingsManager, "logger" | "getSourc
     }
   });
 
+  router.post("/boolean-filter", async (req: Request, res: Response) => {
+    function getQuery(essence: Essence, dimension: Dimension, timekeeper: Timekeeper): Expression {
+      const { dataCube } = essence;
+      const filterExpression = essence
+        .getEffectiveFilter(timekeeper, { unfilterDimension: dimension })
+        .toExpression(dataCube);
+
+      return $("main")
+        .filter(filterExpression)
+        .split(dimension.expression, dimension.name);
+    }
+
+    try {
```

Comment:
When you finish all endpoints, look at the similarities in this try-catch blocks. It would be nice to eliminate duplicated code (extract shared functionality).

Reply:
Definitely! Right now I'm not sure if I want to compose the same functions over and over again or use some pattern a'la middleware for common code ...